HangmanGame/ViewController.swift

```diff
@@ -8,12 +8,165 @@
+        words.shuffle()
+        loadLevel()
+        navigationItem.leftBarButtonItem = UIBarButtonItem(barButtonSystemItem: .refresh, target: self, action: #selector(loadLevel))
+        navigationController?.isToolbarHidden = false
+        
+    }
+    
+    private func divideLetters() {
+        selectedWord = words[1]
+        numberOfLetters = selectedWord.count
+        letters = Array(selectedWord)
+        questionMark.removeAll()
+        correctWord.text = ""
+        for index in 0...numberOfLetters-1{
+            questionMark.append("?")
+            correctWord.text! += questionMark[index]
+        }
+    }
+    
+    private func hideImages() {
+        imgHead.isHidden = true
+        imgBody.isHidden = true
+        imgLeftArm.isHidden = true
+        imgRightArm.isHidden = true
+        imgLeftLeg.isHidden = true
+        imgRightLeg.isHidden = true
+    }
+    
+    private func clearFields() {
+        lblWrongGuess.text = ""
+    }
+    
+    private func showImages() {
+        switch(counter) {
+        case 1:
+            imgHead.isHidden = false
+            break
+        case 2:
+            imgBody.isHidden = false
+            break
+        case 3:
+            imgLeftArm.isHidden = false
+            break
+        case 4:
+            imgRightArm.isHidden = false
+            break
+        case 5:
+            imgLeftLeg.isHidden = false
+            break
+        case 6:
+            imgRightLeg.isHidden = false
+            gameOver()
+            break
+        default:
+            break
+        }
+    }
+    
+    private func alertMsg(title: String, message: String) {
+        let alert = UIAlertController(title: title, message: message, preferredStyle: .alert)
+        alert.addAction(UIAlertAction(title: "OK", style: .default, handler: nil))
+        self.present(alert, animated: true, completion: nil)
+    }
+    
+    private func gameOver() {
+        score -= 1
+        alertMsg(title: "Game Over", message: "You have lost 1 point")
+        loadLevel()
+    }
+    
+    @objc private func loadLevel() {
+        clearFields()
+        hideImages()
+        words.shuffle()
+        divideLetters()
+    }
+    private func victory() {
+        alertMsg(title: "Victory", message: "You've got 1 point")
+        score += 1
+    }
+    
+    @IBAction func btnTryClicked(_ sender: UIButton) {
+        if textBox.text == ""{
+            alertMsg(title: "Error", message: "Please type something")
+        } else {
+            guard var typed = textBox.text?.uppercased() else { return }
+            if typed.count > 1 {
+                alertMsg(title: "Too many arguments", message: "Please type only 1 character")
+            } else {
+                if selectedWord.contains(typed) {
+                    for index in 0...numberOfLetters-1 {
+                        if typed == "\(letters[index])"{
+                            questionMark[index] = typed
```

Comment:
Haven't read up the code in great details, but not seeing guaranteed 1:1 correlation between `numberOfLetters` and size of the array `questionMark`, if it's then I am good. But if it's not, then you should always be aware of potential Array out of bound access, i.e. ensuring `index < questionMark.count`

Reply:
actually I _think_ I see the correlation.